developer/doc_development/doc_development.tex

```diff
@@ -0,0 +1,461 @@
+    \item \alert{No need to index the files}, just put them in the appropriate subfolder.
+    \item You can use the \alert{docstring} at the start of the file not only to write the title of the example, but also the introductory text. This docstring is the only part of the script that is interpreted as RST code without a \alert{\texttt{\#}} at the start of every line.
+    \item \alert{Hypertext links} help users connect various aspects of the library:
+    \begin{itemize}
+        \item \texttt{:class:`\textasciitilde.openturns.ClassName`} to link to the API doc of a class.
+        \item \texttt{:meth:`\textasciitilde.openturns.ClassName.method`} to link to the API doc of a class method.
+        \item \texttt{:ref:`\_theory\_page`} to link to a Theory doc page using its RST label.
+        \item \texttt{:doc:`/auto\_category/subcategory/plot\_xxx`} (no \texttt{.py}) to link to another example.
+    \end{itemize}
+\end{itemize}
+\end{frame}
+
+
+
+%%%%%%%%%%%%%%%%%%%%%
+% API (user manual) %
+%%%%%%%%%%%%%%%%%%%%%
+\section{API (user manual)}
+
+\againframe{tableofcontents}
+
+
+\begin{frame}
+\frametitle{SWIG: Simplified Wrapper and Interface Generator}
+\begin{block}{A tool to link C/C++ libraries with script languages}
+    SWIG parses the library headers (\texttt{.hxx}) and swig interface files (\texttt{.i}) to
+    generate the corresponding module source that must be compiled to
+    produce a binary Python module.
+\end{block}
+\centering \resizebox{!}{4cm}{\includegraphics{swig.png}} \medskip
+
+Python \alert{docstrings} are written in \texttt{\_doc.i.in} files which supplement the \alert{SWIG} \texttt{.i} files.
+
+These docstrings are then read by \alert{Sphinx} to build the API doc section.
+\end{frame}
+
+
+\begin{frame}{Looking up the sources of docstrings}
+Let us consider the \alert{LinearCombinationFunction} class:
+\url{https://openturns.github.io/openturns/master/user_manual/_generated/openturns.LinearCombinationFunction.html}
+
+Corresponding source file:
+\href{https://github.com/openturns/openturns/blob/master/python/src/LinearCombinationFunction_doc.i.in}{python/\alert{src}/LinearCombinationFunction\_doc.i.in}
+
+\begin{itemize}
+    \item Docstring sources are in \texttt{src} rather than \texttt{doc} because they are part of the library.
+    \item A docstring is just a string \texttt{"..."}: double quotes are thus forbidden inside, use \texttt{'...'} instead (not to be confused with \texttt{`...`} as in \texttt{:math:`...`}). Do not put any whitespace between the opening \texttt{"} and the first docstring character, nor between the last docstring character and the closing \texttt{"}.
+    \item Docstrings follow \href{https://numpydoc.readthedocs.io/en/latest/index.html}{\alert{Numpydoc}} style.
+    \item Example code lines start with \texttt{>{}>{}>} and are run during Python tests! If you call \texttt{print}, you need to specify the expected output or the test will fail.
```

Comment:
It sometimes happen that the numerical values in the examples are not the same across platforms, which make the continuous integration machine to fail. In order to make the tests pass, we do not print the digits which are not accurate and replace them with an ellipsis:
```
>>> print(chaosSI.getSobolIndex(0))
0.340236...
```